core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -579,6 +579,7 @@ public RelSubset register(
           "equivRel rowtype",
           equivRel.getRowType(),
           Litmus.THROW);
+      equivRel = ensureRegistered(equivRel, null);
```

Comment:
Should it already be registered? Why need this line?

Reply:
It usually is, but I believe it is not specifically written as required in the api. I think it is safer to add this line just in case some rule feed in an equivalent which is also not yet registered.